src/Database/TigerBeetle/Internal/FFI/BitFlag.hs

```diff
@@ -0,0 +1,36 @@
+{-# LANGUAGE TypeApplications #-}
+module Database.TigerBeetle.Internal.FFI.BitFlag where
+
+import Data.Set qualified as S
+import Data.Set (Set)
+import Data.Bits
+
+flagsToBitmask :: forall a b. (Enum a, FiniteBits b, Num b) => Set a -> b
+flagsToBitmask = S.foldr (\flag acc -> acc .|. fromIntegral (fromEnum flag)) 0
+
+safeFlagsToBitmask :: forall a b. (Bounded a, Enum a, FiniteBits b, Num b) => Set a -> Maybe b
+safeFlagsToBitmask s =
+  let enoughBitsForFlags = S.size s <= finiteBitSize @b 0
+      wordIsBigEnough = fromEnum (maxBound @a) <= 2 ^ finiteBitSize @b 0
+  in if enoughBitsForFlags && wordIsBigEnough
+    then Just $ flagsToBitmask s
+    else Nothing
```

Comment:
These safe functions are mainly intended to sign post the invariants that we expect to hold for the kinds of enum instances that are permissible.